gpiozero/spi_devices.py

```diff
@@ -72,10 +111,10 @@ class AnalogInputDevice(SPIDevice):
     .. _analog to digital converters: https://en.wikipedia.org/wiki/Analog-to-digital_converter
     """
 
-    def __init__(self, bits=None, **spi_args):
-        if bits is None:
-            raise InputDeviceError('you must specify the bit resolution of the device')
+    def __init__(self, bits, **spi_args):
         self._bits = bits
+        self._min_value = -(2 ** bits)
+        self._range = 2 ** (bits + 1) - 1
```

Comment:
Over-simplifying. The *effective* range for pseudo-differential and single-ended ADCs is indeed 0..(2**bits)-1 but given that's a perfect subset of the range -2**bits..(2**bits)-1 we can just treat it as the same range and assume they'll never give us a negative value to normalize (which they won't because there's no 2's complement handling in them).